packages/shell-api/src/database.ts

```diff
@@ -817,13 +817,36 @@ export default class Database extends ShellApiWithMongoClass {
       pipeline.push({ $match: matchingFilters });
     }
 
-    return (await this
-      .getSiblingDB('admin')
-      .aggregate(pipeline, {
-        '$readPreference': {
-          'mode': 'primaryPreferred'
-        }
-      })).toArray();
+    let result;
+
+    try {
+      const cursor = await this
+        .getSiblingDB('admin')
+        .aggregate(pipeline, {
+          $readPreference: {
+            mode: 'primaryPreferred'
+          }
+        });
+      result = await cursor.toArray();
+    } catch (error) {
+      if ((error as any).codeName === 'FailedToParse') {
+        const deepCopyPipeline = JSON.parse(JSON.stringify(pipeline));
+        delete deepCopyPipeline[0].$currentOp.truncateOps;
+
+        const cursor = await this
+          .getSiblingDB('admin')
+          .aggregate(deepCopyPipeline, {
+            $readPreference: {
+              mode: 'primaryPreferred'
+            }
+          });
```

Comment:
Maybe `{ $readPreference: { mode: 'primaryPreferred' } }` could be stored in some `aggregateOptions` variable and then passed to both calls? Similarly `this.getSiblingDB('admin')` could be stored in e.g. `adminDb` to deduplicate them?